content/waku/2024-gantt.md

```diff
@@ -0,0 +1,97 @@
+        (testing) direct messages: milestone, after test-direct-msg, 0
+        Review connection management: crit, milestone, after review-conn-mgmt-1 review-conn-mgmt-2, 0
+        (telemetry) Fleet logging: milestone, after telem-fleet-logging, 0
+        (telemetry) direct message reliability: milestone, after telem-d-msg-rel, 0
+        Reliability Protocol for Relay: crit, milestone, after rel-relay-g rel-relay-n, 0
+        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-g rel-reqres-j, 0
+        Review MVDS usage and fail path: crit, milestone, after mvds, 0
+    section Static Sharding - dedicated shards
+        (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+
+%% Tasks
+    section golang eng 1
+        %% Estimation TBC
+        Review connection management: crit, review-conn-mgmt-1, 2024-05-13, 8w
+        %% Estimate TBC
+        Review MVDS usage and fail path: crit, mvds, after review-conn-mgmt-1, 6w
+        %% Estimate/start TBC
+        (testing) direct messages: test-direct-msg, after mvds, 4w
+        %% Estimate TBC
+        (testing) Custom shard impl of Communities: test-custom-shard, after test-direct-msg, 4w 
+    section golang eng 2
+        Reliability Protocol for Relay (status-go): crit, rel-relay-g, 2024-05-13, 12w
+    section golang eng 3
+        Store v3 (go-waku client only): crit, storev3-g, 2024-02-26, 2024-05-24
+        %% Estimate TBC - assuming parallel work possible
+        Review connection management: crit, review-conn-mgmt-2, after storev3-g, 8w
+        %% TBC Estimate
+        Sharding peer mgmt and discovery hardening (go-waku): crit, sh-peer-mgmt-g, after review-conn-mgmt-2, 8w
+    section golang eng 4
+        %% Review if actually needed
+        Store v3 (sync): crit, 2024-02-08, 2024-04-26
+        %% Estimate TBC assuming no RFC here
+        Reliability Protocol for Resource-Restricted Clients (status-go): crit, rel-reqres-g, 2024-05-13, 10w
+        (metric) Count store messages: count-store-msg, after rel-reqres-g, 2w
+    section golang eng 5
+        %% Estimate TBD
+        DoS protection for req-res protocols (go-waku client only): crit, dosreqresg, 2024-05-20, 4w
+        %% TBC estimate and start
+        (telemetry) direct message reliability: telem-d-msg-rel, 2024-06-17, 6w
+        %% TBC estimate and start
+        (telemetry) Measure Bandwidth: telem-bandwidth, after telem-d-msg-rel, 8w
+    section test eng 1
+        %% TODO: start small scale simulations of DST such store perf    
+    section nim eng 1
+        %% TBC estimation done?
+        Store v3-beta (msg hash): crit, storev3-br, 2024-01-01, 12w
+        %% TBC estimation
+        Store v3 (sync): crit, storev3-r, after storev3-br, 20w
```

Comment:
I think it's realistic to expect research work could be done by end Q2 (~30 June). Note that this includes the reference implementation in Nim, a specification for Store Sync and at least some real-world testing. cc @SionoiS @chaitanyaprem - too ambitious given current state?

Reply:
Waku Sync in it's simplest form can be considered "done". Testing is next step and after that, improvement will depends on our findings. In parallel spec. can be done and since we use another protocol should be fast to write.

As for other features forget about it if we want to make the Q2 deadline.

Reply:
> What features are we talking about?

Allowing other assumptions than sync the last hour of messages.

Reply:
Indeed. I don't think we need these features for the Status roadmap. The Store Sync in simplest form can therefore be done within Q2. I think we should be realistic and expect a number of unknowns cropping up during testing.